mungegithub/reports/shame.go

```diff
@@ -0,0 +1,280 @@
+		}
+		if len(strs) == 0 {
+			needsIndividualEmail[assignee] = true
+			continue
+		}
+		sort.Strings(strs)
+		chunks = append(chunks, fmt.Sprintf("%v:\n%v", assignee, strings.Join(strs, "\n")))
+	}
+	sort.Strings(chunks)
+
+	// Gather addresses
+	to := []string{}
+	missingAddresses := []string{}
+	for u, e := range r.loginToEmail {
+		if mayEmail(e) {
+			to = append(to, e)
+		} else {
+			missingAddresses = append(missingAddresses, u)
+		}
+	}
+	sort.Strings(to)
+	sort.Strings(missingAddresses)
+
+	dest := &bytes.Buffer{}
+
+	// Write the report
+	if s.From != "" {
+		fmt.Fprintf(dest, "From: %v\n", s.From)
+	}
+	if s.Cc != "" {
+		fmt.Fprintf(dest, "Cc: %v\n", s.Cc)
+	}
+	fmt.Fprintf(dest, "To: %v\n", strings.Join(to, ","))
+	fmt.Fprintf(dest, "Subject: Kubernetes flaky Test Report: %v flaky tests\n", r.totalTests)
+	fmt.Fprintf(dest, `
+If you are in the To: line of this email, you have flaky tests to fix! Flaky
+tests, even if they flake only a small percentage of the time, cause the merge
+queue to become very long, which causes everyone on the team pain and
+suffering.  Please either fix the tests assigned to you or find them an owner
+who will fix them.
+
+There were %v P2/P3 issues which are not reported here.
+
+Full report:
+`, r.lowPriorityTests)
+	fmt.Fprintf(dest, "%s\n", strings.Join(chunks, "\n\n"))
+	if len(missingAddresses) > 0 {
+		fmt.Fprintf(dest, `
+These users couldn't be added to the To: line, as we have no address for them:
+
+%v
+
+Individuals with an accessible email and no assignments older than 3 days will
+be left off the group email, so please make your email address public in github!
+
+Note: non-google users are not emailed by this system.
+
+`, strings.Join(missingAddresses, ", "))
+	}
+
+	return needsIndividualEmail, s.runCmd(dest)
+}
+
+func (s *ShameReport) individualReport(user string, r *reportData) error {
+	// Gather report body
+	issues := r.loginToIssues[user]
+	strs := []string{}
+	for _, data := range issues {
+		strs = append(strs, data.String())
+	}
+	sort.Strings(strs)
+	chunk := fmt.Sprintf("%v:\n%v\n", user, strings.Join(strs, "\n"))
+
+	to := []string{}
+	email := r.loginToEmail[user]
+	if mayEmail(email) {
+		to = append(to, email)
+	}
+	sort.Strings(to)
+
+	dest := &bytes.Buffer{}
+
+	// Write the report
+	if s.From != "" {
+		fmt.Fprintf(dest, "From: %v\n", s.From)
+	}
+	// No Cc on individual emails!
+	fmt.Fprintf(dest, "To: %v\n", strings.Join(to, ","))
+	fmt.Fprint(dest, "Bcc: dbsmith@google.com\n") // so I know it works
+	fmt.Fprintf(dest, "Subject: Kubernetes flaky Test Report: %v flaky tests\n", r.totalTests)
+	fmt.Fprintf(dest, `
+Hi %v,
+
+This is a note to let you know that you have flaky tests assigned to you.
+Owners of tests broken for less than 3 days are left off the group email!
+
+Full report:
+`, user)
+	fmt.Fprint(dest, chunk)
+
+	return s.runCmd(dest)
+}
+
+func mayEmail(email string) bool { return strings.HasSuffix(email, "@google.com") }
```

Comment:
Well, for one, I won't get in trouble for doing it ;)

Honestly this gets most of the issues right now. I should let other orgs/people opt-in, maybe.